cyclonedx-bom/src/models/metadata.rs

```diff
@@ -41,6 +44,15 @@ pub struct Metadata {
 }
 
 impl Metadata {
+    /// Constructs a new `Metadata` with a timestamp of DateTime::now()
+    /// ```
+    /// use cyclonedx_bom::models::metadata::Metadata;
+    ///
+    /// let metadata = Metadata::new();
+    /// ```
+    /// # Errors
+    ///
+    /// Returns an error variant if unable to generate a valid timestamp based on the current time
```

Comment:
Also wondering if this could use a better description. Based off the Rust API doc style guide.